Shared/PaymentViewModel.swift

```diff
@@ -0,0 +1,83 @@
+//
+//  PaymentViewModel.swift
+//  MobilePay-Apple
+//
+//  Created by Momo Ozawa on 2021/07/19.
+//
+
+import Foundation
+import StoreKit
+import MobilePayKit
+
+class PaymentViewModel: NSObject, ObservableObject {
+
+    @Published var contentList: [PurchasableContent] = []
+
+    @Published var currentContent: PurchasableContent?
+
+    private let paymentQueueService = PaymentQueueService()
+
+    private let productIdentifiers = Set([
+        "com.mobilepay.consumable.rocketfuel",
+        "com.mobilepay.consumable.premiumrocketfuel"
+    ])
+
+    // Here we're going to store all the completed purchases
+    private var completedPurchases = [String]() {
+
+        // We need to update set the subscription.isLocked value to true after a purchase
+        didSet {
+            // We have to do this on the main queue as this might affect the UI
+            DispatchQueue.main.async { [weak self] in
+                // Ensure we can access self here
+                guard let self = self else { return }
```

Comment:
```suggestion
                guard let self = self else {
                  return
                }
```

Just to conform with our standards + removing the comment.

Reply:
Fixed! 187c56e